larpandoracontent/LArThreeDReco/LArThreeDBase/ThreeViewTrackMatchingAlgorithm.h

```diff
@@ -1,42 +1,36 @@
 /**
- *  @file   larpandoracontent/LArThreeDReco/LArThreeDBase/ThreeDTracksBaseAlgorithm.h
+ *  @file   larpandoracontent/LArThreeDReco/LArThreeDBase/ThreeViewTrackMatchingAlgorithm.h
  *
- *  @brief  Header file for the three dimensional tracks algorithm base class.
+ *  @brief  Header file for the three view track matching algorithm class.
  *
  *  $Log: $
  */
-#ifndef LAR_THREE_D_TRACKS_BASE_ALGORITHM_H
-#define LAR_THREE_D_TRACKS_BASE_ALGORITHM_H 1
+#ifndef LAR_THREE_VIEW_TRACK_MATCHING_ALGORITHM_H
+#define LAR_THREE_VIEW_TRACK_MATCHING_ALGORITHM_H 1
 
 #include "larpandoracontent/LArObjects/LArTwoDSlidingFitResult.h"
 
-#include "larpandoracontent/LArThreeDReco/LArThreeDBase/ThreeDBaseAlgorithm.h"
-
-#include <unordered_map>
+#include "larpandoracontent/LArThreeDReco/LArThreeDBase/ThreeViewMatchingAlgorithm.h"
 
 namespace lar_content
 {
 
-typedef std::unordered_map<const pandora::Cluster*, pandora::CartesianPointVector> SplitPositionMap;
-
-//------------------------------------------------------------------------------------------------------------------------------------------
-
 /**
- *  @brief  ThreeDTransverseTracksAlgorithm class
+ *  @brief  ThreeViewTrackMatchingAlgorithm class
  */
 template<typename T>
-class ThreeDTracksBaseAlgorithm : public ThreeDBaseAlgorithm<T>
+class ThreeViewTrackMatchingAlgorithm : public ThreeViewMatchingAlgorithm<T>
```

Comment:
I think the matching classes can be reduced to a single instance of TrackMatchingAlgorithm<T, U> -> MatchingAlgorithm<T, U> -> MatchingBaseAlgorithm with the addition of the extra type in the template specification. I think a new abstract class, ClusterOverlapContainer (or some better name), could take ownership of the cluster lists, input cluster lists and the overlap tensor/matrix, along with the functions to perform the UpdateForNewCluster, UpdateUponDeletion, GetClusterListName etc. which can handle the specific tasks relating to 2 or 3 views.

Most of the other functions then become essentially identical for the 2 or 3 view case, so you wouldn't need Two and Three view classes for matching, just a single hierarchy where all of the differences relating to the number of views get moved to this new mini class hierarchy.